platform/lambda/lambda.go

```diff
@@ -227,8 +222,28 @@ func (p *Platform) CreateStack(region, version string) error {
 	return stack.New(p.config, p.events, region).Create(version)
 }
 
+// UpdateStack implementation.
+func (p *Platform) UpdateStack(region, version string) error {
+	return stack.New(p.config, p.events, region).Update(version)
+}
+
+// SetS3BucketName based on stack.
+func (p *Platform) SetS3BucketName(region string) error {
+	return stack.New(p.config, p.events, region).SetS3BucketName()
```

Comment:
How come we need this stack method? Maybe we should generate some random bucket name like `<app-name>-<uuid>` since it's just for Lambda uploads, the user likely doesn't even care to know that we happen to use S3.

That should simplify the code below. I suppose we'd have to store it somewhere on initial creation though, hmmm...

Reply:
If the `up` stack is being create for the first time this PR delegates to aws cloudformation the s3 bucket name generation by creating a new cloudformation stack with just a s3 resource (using a `Logical ID`). Then it gets the generated s3 bucket name (`Physical ID`) to upload the function using s3manager and finally updates the stack with the another resources.

If the stack already exists, just sets the s3 bucket name into `config.S3BucketName` by getting its name from `DescribeStackResource`.

That said, maybe we could simplify those steps by generating the s3 bucket in `up` instead of delegating it to aws cloudformation, is that you are saying @tj?

Reply:
hmm yeah I need to think on that a bit, both cases kind of suck I guess if we're introducing more latency/requests there.

Reply:
Yeah, I'm not sure we need to delegate the s3 bucket name to aws cloudformation. Maybe something like `up-deploy-${service-name}` is enough.
This will save a lot of requests.

Reply:
agreed! that would definitely help keep things cleaner

Reply:
Ok, I'm going to implement this way then